library/template.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package library
+
+import (
+	"fmt"
+)
+
+// Template is the library representation of a template for a pipeline.
+//
+// swagger:model Template
+type Template struct {
+	HTMLURL *string `json:"id,omitempty"`
```

Comment:
I think we want to fix the `json` tag here 👍 

```suggestion
	HTMLURL *string `json:"html_url,omitempty"`
```

Also, thoughts on calling it `URL` vs `HTMLURL`?

Thinking this more closely matches what we've declared in other structs:

https://github.com/go-vela/types/blob/226d0cc72538872fc5f8f144c7d380b7ca3b4d11/library/deployment.go#L9-L22

Reply:
went with HTMLURL to be consistent with the git payload field being `html_url`. Technically Source is a URL too? I can change it, no biggie

Reply:
https://developer.github.com/v3/repos/contents/#get-repository-content see payloads

I like URL though. Just thought it will make more sense because the function to fetch this data will be called source.GetHTMLURL()

Reply:
updated from `HTMLURL`/`html_url` to  `URL`/`url`

Reply:
I hear ya 👍 

I think the real difference is that `Source` wouldn't be a fully qualified url because it's missing the protocol/scheme (`http://`, `https://`).

Reply:
agreed, good point, thank you 👍